frontend/src/components/FamilySignUpModal.tsx

```diff
@@ -0,0 +1,107 @@
+// import React, { useEffect, useState } from 'react';
+import React, { useContext } from 'react';
+import {
+    Modal,
+    ModalOverlay,
+    ModalContent,
+    ModalCloseButton,
+    Button,
+    Divider,
+    Heading,
+    ModalFooter,
+    SimpleGrid,
+
+} from '@chakra-ui/react';
+import moment from 'moment';
+import { Member } from '../../../src/typedefs/member';
+import { Job, PatchJobRequest } from '../../../src/typedefs/job';
+import { UserContext } from '../contexts/UserContext';
+
+interface modalProps {
+  isOpen: boolean,
+  onClose: () => void,
+  familyMembers: Member[],
+  selectedEvent: any, // TODO: bugfix needed, Event | Job breaks here
+  // eslint-disable-next-line no-unused-vars
+  signUpForJob: (patchInfo: { jobId: number; editedJob: PatchJobRequest; }) => void
+}
+
+export default function FamilySignUpModal(props: modalProps) {
+    const { state } = useContext(UserContext);
+
+    function isJob(selectedEvent: Event | Job): selectedEvent is Job {
+        if ((selectedEvent as Job).jobId) {
+            return true;
+        }
+        // else, its an Event
+        return false;
+    }
+
+    async function generateJobSignUpPatch(memberId: number) {

```

Comment:
if this does not depend on local state, move it outside the function. If it does. wrap it in `useCallback(() => ... ,[states])`